### ISSUE TITLE
Hitting ^C inside the shell causes stack trace in the frontend and orphaned tasks

### ISSUE THREAD

**Issue by cs-shadow** [cs-shadow/buildstream#35]
In GitLab by @tristanvb on May 18, 2017, 07:14

As the description says, interrupt in the shell (I think depending on what shell you are using in the sandbox, this happens with dash but I'm not sure it happens with bash) will cause a stack trace in the buildstream frontend, exit with some failure and leave ongoing tasks orphaned.

**Comment by cs-shadow** [comment]
In GitLab by @tristanvb on May 30, 2017, 10:12

closed via commit a0b0de9a2351d2fb47b325fdc0cc07e0058b7b50

**Comment by cs-shadow** [comment]
In GitLab by @tristanvb on May 30, 2017, 10:12

Fixed with the following commit:

```
commit a0b0de9a2351d2fb47b325fdc0cc07e0058b7b50
Author: Tristan Van Berkom <tristan.vanberkom@codethink.co.uk>
Date:   Tue May 30 19:09:59 2017 +0900

    _sandboxbwrap.py: Fix issue #31, handle keyboard interrupts
    
    When running an interactive shell that does not have job control
    available, keyboard interrupts will be propagated to the parenting
    buildstream process.
    
    Just handle KeyboardInterrupt here and ensure that the sandbox
    exited cleanly.

```
